### PR TITLE
feat: Refactor Router logic

### PR DESCRIPTION
**Pull request by notJoon**
# Description

- Reduce the complexity of `SwapRoute` (complexity score: 24 -> under 10)
- Reduce the complexity of `DrySwapRoute` (score: 21 -> under 10)

### CODE REVIEW

**Review by dongwon8247 — COMMENTED** [review]
Docs seem to be improved a lot, but please double check with @r3v4s if this change doesn't effect to the actual logic. Otherwise good to go!

File: router/router.gno [hunk 1]
```diff
@@ -189,37 +166,50 @@ func handleSingleSwap(route string, amountSpecified *i256.Int, isDry bool) (*u25
 	return singleSwap(singleParams)
 }
 
+func finalizeSwap(inputToken, outputToken string, resultAmountIn, resultAmountOut *u256.Uint, swapType string, tokenAmountLimit *u256.Uint, userOldWugnotBalance uint64) (string, string) {
+    afterFee := handleSwapFee(outputToken, resultAmountOut, false)
+
+	userNewWugnotBalance := wugnot.BalanceOf(a2u(std.GetOrigCaller()))
```
Comment (r3v4s) [inline router/router.gno:172]: lint problem, I think

**Review by r3v4s — CHANGES_REQUESTED** [review]
tests fails on my local env. Please check from your env

File: router/router.gno (hunk 1 above)
Comment (notJoon) [reply router/router.gno]: https://github.com/gnoswap-labs/gnoswap/pull/297/commits/d034004817c79830272c0af01a37a3496f4bbdd2

**Comment by notJoon** [comment]
> tests fails on my local env. Please check from your env

fixed https://github.com/gnoswap-labs/gnoswap/pull/297/commits/4434c47bb7ee4cb96e5cb5833d5a6249c8d0cc42

**Review by r3v4s — APPROVED** [review]
IMHO, high change of conflict if you fix ongoing position constructor issue.

File: router/router.gno [hunk 2]
```diff
@@ -204,7 +207,7 @@ func handleMultiSwap(swapType string, route string, numHops int, amountSpecified
 		if isDry {
 			return multiSwapDry(swapParams, 0, numHops, route) // iterate here
 		}
-		return multiSwap(swapParams, 0, numHops, route) // iterate here
+		return multiSwap(*sp, 0, numHops, route) // iterate here
```
Comment (r3v4s) [inline router/router.gno:210]:
```suggestion
		return multiSwap(swapParams, 0, numHops, route) // iterate here
```

**Review by r3v4s — APPROVED** [review]

**Review by r3v4s — APPROVED** [review]

**Review by dongwon8247 — APPROVED** [review]

**Pull request merged**